src/Google.Api.Gax/Api/Gax/RetrySettings.cs

```diff
@@ -0,0 +1,155 @@
+            GaxPreconditions.CheckArgument(RetryFilter != null, paramName, "No retry filter specified");
+            return this;
+        }
+
+        /// <summary>
+        /// Provides a mechanism for applying jitter to delays between retries.
+        /// See the <see cref="DelayJitter"/> property for more information.
+        /// </summary>
+        public interface IJitter
+        {
+            /// <summary>
+            /// Returns the actual delay to use given a maximum available delay.
+            /// </summary>
+            /// <param name="maxDelay">The maximum delay provided by the backoff settings</param>
+            /// <returns>The delay to use before retrying.</returns>
+            TimeSpan GetDelay(TimeSpan maxDelay);
+        }
+
+        /// <summary>
+        /// The default jitter which returns a uniformly distributed random delay between 0 and
+        /// the specified maximum.
+        /// </summary>
+        public static IJitter RandomJitter { get; } = new RandomJitterImpl();
+
+        /// <summary>
+        /// A jitter which simply returns the specified maximum delay.
+        /// </summary>
+        public static IJitter NoJitter { get; } = new NoJitterImpl();
+
+        // TODO: Is this a reasonable default? Does it make sense to have a default at all?
+
+        /// <summary>
+        /// The default retry filter, which retries operations which fail due to a status code of "not found".
+        /// </summary>
+        public static Predicate<RpcException> DefaultFilter { get; } = FilterForStatusCodes(StatusCode.NotFound);
+
+        /// <summary>
+        /// Creates a retry filter based on status codes.
+        /// </summary>
+        /// <param name="statusCodes">The status codes to retry. Must not be null.</param>
+        /// <returns>A retry filter based on status codes.</returns>
+        public static Predicate<RpcException> FilterForStatusCodes(params StatusCode[] statusCodes) =>
+            FilterForStatusCodes((IEnumerable<StatusCode>) statusCodes);
+
+        /// <summary>
+        /// Creates a retry filter based on status codes.
+        /// </summary>
+        /// <param name="statusCodes">The status codes to retry. Must not be null.</param>
+        /// <returns>A retry filter based on status codes.</returns>
+        public static Predicate<RpcException> FilterForStatusCodes(IEnumerable<StatusCode> statusCodes)
+        {
+            GaxPreconditions.CheckNotNull(statusCodes, nameof(statusCodes));
+            // TODO: Take a copy? Optimize for common cases?
+            return ex => statusCodes.Contains(ex.Status.StatusCode);
+        }
+
+        private sealed class RandomJitterImpl : IJitter
+        {
+            private static int seed = Environment.TickCount;
+
+            private static ThreadLocal<Random> randomWrapper = new ThreadLocal<Random>(() =>
+                new Random(Interlocked.Increment(ref seed))
```

Comment:
It means that if lots of instances are created in very swift succession, we don't end up with them all having the same seed.